shop/src/Components/ShopBy/index.jsx

```diff
@@ -1,23 +1,72 @@
 import Price from "../Price";
-import CheckboxComponent from "../CheckboxComponent";
+import "./ShopBy.scss";
 import colors from "../../Data/colors.json";
-import {sizes} from "../../Data/sizeAndSortList.js";
+import { sizes } from "../../Data/sizeAndSortList.js";
 import brands from "../../Data/brands.json";
-import "./ShopBy.scss";
-const ShopBy = () => {
+import CheckboxComponent from "../CheckboxComponent";
+import { useState } from "react";
+
+const ShopBy = ({
+  shoeItems,
+  handleOnPriceRange,
+  handleOnGetItemFromCheckbox,
+}) => {
+  const [allCheckedItem, setAllCheckedItem] = useState([]);
+
+  const checkboxResult = shoeItems.filter((shoe) => {
+    if (allCheckedItem.length === 0) return shoeItems;
+    return allCheckedItem.find(
+      (col) =>
+        shoe.color === col ||
+        allCheckedItem.find(
+          (designer) =>
+            shoe.designer === designer ||
+            allCheckedItem.find(() => shoe.size_range.includes(20))
+        )
+    );
+  });
+
+  const handleOnCheck = (e) => {
+    const value = e.target.value;
+    const checked = e.target.checked;
+
+    let itemsChecked = [...allCheckedItem];
+    if (checked) {
+      itemsChecked = [...allCheckedItem, value];
+    } else {
+      itemsChecked.splice(allCheckedItem.indexOf(value), 1);
+    }
+    setAllCheckedItem(itemsChecked);
+
+    handleOnGetItemFromCheckbox?.([...checkboxResult]);
+  };
+
   return (
     <div className="shopBy">
       <div className="sortBy">
         <h3>Shop By</h3>
       </div>
       <div>
-        <Price />
+        <Price
+          handleOnPriceRange={handleOnPriceRange}
+          shoeItems={shoeItems}
+          step={1}
```

Comment:
what is this `1`

Reply:
It is each step on input type range. it increase or decrease once 1.